README.md

```diff
@@ -1,233 +1,249 @@
-$ yarn test:functional
-```
+### Route Tests
+We test all routes contained within the service's [routes file](./src/main/routes/routes.ts).
 
-### Security
+Route testing config can be found within the [jest.routes.config.js](./jest.routes.config.js).
 
-#### CSRF prevention
+Route tests are run using `yarn test:routes`
 
-[Cross-Site Request Forgery](https://github.com/pillarjs/understanding-csrf) prevention has already been
-set up in this template, at the application level. However, you need to make sure that CSRF token
-is present in every HTML form that requires it. For that purpose you can use the `csrfProtection` macro,
-included in this template app. Your njk file would look like this:
+### Accessibility (a11y) Tests
 
-```
-{% from "macros/csrf.njk" import csrfProtection %}
-...
-<form ...>
-  ...
-    {{ csrfProtection(csrfToken) }}
-  ...
-</form>
-...
-```
+We test the accessibility of all routes contained within the service's [routes file](./src/main/routes/routes.ts) using [`pa11y`](https://pa11y.org/) and [`Supertest`](https://github.com/visionmedia/supertest).
 
-#### Helmet
+Accessibility testing configuration can be found within the [`jest.a11y.config.js`](./jest.a11y.config.js) file.
 
-This application uses [Helmet](https://helmetjs.github.io/), which adds various security-related HTTP headers
-to the responses. Apart from default Helmet functions, following headers are set:
+To run the a11y tests, use the `yarn test:a11y` command.
 
--   [Referrer-Policy](https://helmetjs.github.io/docs/referrer-policy/)
--   [Content-Security-Policy](https://helmetjs.github.io/docs/csp/)
+### Functional (E2E) Tests
 
-There is a configuration section related with those headers, where you can specify:
+We currently use two forms of functional tests which both run in the build pipeline. Legacy tests (of which there are many) were written using [`Webdriver.io (WDIO)`](https://webdriver.io/) and new tests using [`CodeceptJS`](https://codecept.io/).
+We plan to migrate to using purely Codecept tests iteratively, slowly deprecating the old style.
 
--   `referrerPolicy` - value of the `Referrer-Policy` header
+#### Justification for migration to CodeceptJS
+- Cross-browser testing.
+- Plugin ecosystem allows smooth implementation of helpers and utilities on existing tests.
+- Rapid test creation due to simple syntax.
+- [Allure reporting](https://docs.qameta.io/allure/) in CodeceptJS allows developers to view detailed information about test cases, including test steps, screenshots, console logs, and error messages. The Allure report can help to identify trends in test results, identify areas of the application that require further testing or improvement, and provide a detailed view of the state of the test suite.
 
-Here's an example setup:
+#### Running e2e tests:
+Our legacy e2e tests can be run using:
 
-```json
-    "security": {
-      "referrerPolicy": "origin",
-    }
-```
+`yarn test:functional` (headless) or `yarn test:functional-dev`
```

Comment:
old legacy test only should be run using 'yarn test:functional:old'. The one mentioned here `yarn test:functional' will run both old and new tests